store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -4868,6 +4868,21 @@ public void deleteDistributionList(String zimbraId) throws ServiceException {
     }
 
     private void deleteDistributionList(LdapDistributionList dl) throws ServiceException {
+        deleteDistributionList(dl, false);
+    }
+
+    private void deleteDistributionList(LdapDistributionList dl, boolean cascadeDelete) throws ServiceException {
+        // check if cascadeDelete is true. If it's true, delete all subgroups.
+        if (cascadeDelete) {
+            Set<String> members = dl.getAllMembersSet();
+            for (String member : members) {
+                LdapDistributionList subDl = (LdapDistributionList) getDistributionListByNameInternal(member);
```

Comment:
This will only delete the direct child members of this group.Direct child group can have other groups as member.Those groups should also be deleted. Have to delete the full hierarchy of groups.

Reply:
deleteDistributionList is called recursively so hierarchy is taken care. or you are saying something else?